### PR TITLE
fix(mixed): avoid usages of global `document` and `window`

### LINKED ISSUE #1964: Remove usage of global document and window from components
## Bug Report

Following components still contain references to global document/window:
- Dropdown.tsx line 814
- Loader.tsx line 105
- Popup.tsx line 276
- Toolbar.tsx lines 182, 189, 191
- ToolbarRadioGroup.tsx line 103

Also, check focusUtilities, lines 430 and 440 which also use the globals, but just as a fallback value.

### Version
0.39.0

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #1964.

This PR removes remaining usages of `document` and `window` globals in components. Also forces ESLint to catch such issues.

### CODE REVIEW

File: packages/react/.eslintrc.json [hunk 1]
```diff
@@ -6,6 +6,16 @@
...
+    },
+    {
+      "files": "src/**/*.{ts,tsx}",
+      "globals": {
+        "document": "off",
+        "window": "off"
+      },
+      "rules": {
+        "no-undef": "error"
+      }
```
Comment (layershifter) [inline packages/react/.eslintrc.json:18]: Temporary solution until we will merge #1963.

File: packages/react/.eslintrc.json [hunk 2]
```diff
@@ -6,6 +6,16 @@
       "rules": {
         "react/prop-types": "off"
       }
+    },
+    {
+      "files": "src/**/*.{ts,tsx}",
+      "globals": {
+        "document": "off",
+        "window": "off"
```
Comment (layershifter) [inline packages/react/.eslintrc.json:14]: It's reasonable to disable only `document` and `window` globals as they are context dependent.

File: packages/react/src/components/Popup/Popup.tsx [hunk 3]
```diff
@@ -273,7 +273,9 @@ export default class Popup extends AutoControlledComponent<PopupProps, PopupStat
     const lastContentRef = getRefs().pop()
     const isLastOpenedPopup: boolean =
       lastContentRef && lastContentRef.current === this.popupDomElement
-    const bodyHasFocus: boolean = document.activeElement === document.body
+
+    const activeDocument = this.props.mountDocument || this.context.target
```
Comment (layershifter) [inline packages/react/src/components/Popup/Popup.tsx:277]: We need to consider remove `mountDocument` prop from `Popup` at all.

File: packages/react/src/components/Toolbar/Toolbar.tsx [hunk 4]
```diff
@@ -179,16 +179,17 @@ class Toolbar extends UIComponent<WithAsProp<ToolbarProps>, ToolbarState> {
...
   }
 
   afterComponentRendered() {
     if (this.animationFrameId !== undefined) {
-      window.cancelAnimationFrame(this.animationFrameId)
+      cancelAnimationFrame(this.animationFrameId)
     }
-    this.animationFrameId = window.requestAnimationFrame(() => {
+
+    this.animationFrameId = requestAnimationFrame(() => {
```
Comment (layershifter) [inline packages/react/src/components/Toolbar/Toolbar.tsx:192]: Discussed with @miroslavstastny, there are no reasons to use it from `window`

**Review by jurokapsiar — APPROVED** [review]

**Pull request merged**